uint8 mode
int16 data
uint8 clear_flag
